File: app/src/main/java/com/possible/architecturesample/data/network/NetworkDataSource.java
package com.possible.architecturesample.data.network;

import com.possible.architecturesample.data.network.responses.BookResponse;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Url;
import rx.Observable;

public interface NetworkDataSource {
    @GET()
    Observable<List<BookResponse>> getObservableBooks(@Url String url);
}

File: app/src/main/kotlin/com/possible/architecturesample/data/network/NetworkDataSource.kt
package com.possible.architecturesample.data.network

import com.possible.architecturesample.data.network.responses.BookResponse

import retrofit2.http.GET
import retrofit2.http.Url
import rx.Observable

interface NetworkDataSource {
    @GET
    fun getObservableBooks(@Url url: String): Observable<List<BookResponse>>
}
